### PR TITLE
minor improvements to client

### PR DESCRIPTION
**Pull request by sg-s**
## changes

-  removed a file that shouldn't be there
- methods now have a `debug` parameter which prints debug information about every API request

### CODE REVIEW

File: src/data_hub/_api.py [hunk 1]
```diff
@@ -111,6 +124,13 @@ def dynamic_function(*, client=None, **kwargs):
             else:
                 raise error
 
+        if isinstance(response, _response.APIResponse):
```
Comment (jonrkarr) [inline src/data_hub/_api.py:127]: Would `if debug:` be simpler?

The meaning of this if condition seems hard to understand. I understand the condition is when response has the type APIResponse. What type APIResponse means vs other possible types could be clearer. It seems like the condition for this is similar to `debug=True` and that this is related to `client = client.with_raw_response`.

File: src/data_hub/_api.py [hunk 2]
```diff
@@ -76,6 +81,9 @@ def _get_default_client(client=None, refresh: bool = True):
             base_url=base_url,
         )
 
+        if debug:
+            client = client.with_raw_response
```
Comment (jonrkarr) [inline src/data_hub/_api.py:85]: This feels like a complicated way to make the behavior conditional on `debug=True`.

**Review by jonrkarr — APPROVED** [review]
Approving because this seems to only affect the behavior for debugging. 

I think conditioning on `if debug:` would make the code easier to understand.

File: src/data_hub/_api.py (hunk 1 above)
Comment (sg-s) [reply src/data_hub/_api.py:127]: This makes sense. I'll change.

Comment (sg-s) [reply src/data_hub/_api.py:127]: fixed

File: src/data_hub/_api.py (hunk 2 above)
Comment (sg-s) [reply src/data_hub/_api.py:85]: fixed

**Pull request merged**